sdk_extensions/logging/support/src/main/java/io/opentelemetry/logging/BatchLogProcessor.java

```diff
@@ -0,0 +1,372 @@
+                  }
+                } catch (InterruptedException | ExecutionException e) {
+                  logger.warning("log exporter failure:" + e.getLocalizedMessage());
+                  exporterFailureCounter.add(1);
+                } catch (TimeoutException e) {
+                  logger.warning("log exporter has failed to return async result");
+                  exporterFailureCounter.add(1);
+                }
+              }
+            }
+          },
+          exporterTimeout);
+    }
+
+    private static List<LogRecord> createLogDataForExport(
+        ArrayList<LogRecord> recordsToShip, int startIndex, int endIndex) {
+      List<LogRecord> logDataBuffer = new ArrayList<>(endIndex - startIndex);
+      for (int i = startIndex; i < endIndex; i++) {
+        logDataBuffer.add(recordsToShip.get(i));
+        recordsToShip.set(i, null);
+      }
+      return Collections.unmodifiableList(logDataBuffer);
+    }
+
+    public void addLogRecord(LogRecord record) {
+      synchronized (monitor) {
+        if (logRecords.size() >= maxQueueSize) {
+          droppedRecordCounter.add(1);
+          // FIXME: call callback
+        }
+        logRecords.add(record);
+        if (logRecords.size() >= maxExportBatchSize) {
+          monitor.notifyAll();
+        }
+      }
+    }
+
+    private void shutdown() {
+      forceFlush();
+      timer.cancel();
+      exporter.shutdown();
+    }
+
+    private void forceFlush() {
+      ArrayList<LogRecord> logsCopy;
+      synchronized (monitor) {
+        logsCopy = new ArrayList<>(this.logRecords);
+        logRecords.clear();
+      }
+      exportBatches(logsCopy);
+    }
+  }
+
+  static class Builder extends ConfigBuilder<Builder> {
+    /* @VisibleForTesting */ static final long DEFAULT_SCHEDULE_DELAY_MILLIS = 5000;
+    static final int DEFAULT_MAX_QUEUE_SIZE = 2048;
+    static final int DEFAULT_MAX_EXPORT_BATCH_SIZE = 512;
+    static final long DEFAULT_EXPORT_TIMEOUT_MILLIS = 30_000;
+
+    private final Exporter exporter;
+    private long scheduleDelayMillis = DEFAULT_SCHEDULE_DELAY_MILLIS;
+    private int maxQueueSize = DEFAULT_MAX_QUEUE_SIZE;
+    private int maxExportBatchSize = DEFAULT_MAX_EXPORT_BATCH_SIZE;
+    private long exporterTimeoutMillis = DEFAULT_EXPORT_TIMEOUT_MILLIS;
+
+    private Builder(Exporter exporter) {
+      this.exporter = Utils.checkNotNull(exporter, "Exporter argument can not be null");
+    }
+
+    public Builder newBuilder(Exporter exporter) {
+      return new Builder(exporter);
+    }
+
+    public BatchLogProcessor build() {
+      return new BatchLogProcessor(
+          maxQueueSize, scheduleDelayMillis, maxExportBatchSize, exporterTimeoutMillis, exporter);
+    }
+
+    /**
+     * Sets the delay interval between two consecutive exports. The actual interval may be shorter
+     * if the batch size is getting larger than {@code maxQueuedSpans / 2}.
+     *
+     * <p>Default value is {@code 5000}ms.
```

Comment:
Can we have a smaller default, please? I don't want to increase logging end-to-end latency this much.